src/check_names.cpp

```diff
@@ -0,0 +1,76 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// jsonify a character vector
+std::string jsonify(CharacterVector x) {
+  std::string out = "[";
+  for (int i = 0; i < x.size(); i++) {
+    out += "\"";
+    out += x[i];
+    out += "\",";
+  }
+  // trim trailing comma
+  if (x.size() > 0) {
+    out.resize(out.size() - 1);
+  }
+  out += "]";
+  return out;
+}
+
+// x is a list of list. verify all sublists have same number of element as
+// column count. verify all sublists have same names if available.
+// [[Rcpp::export(name = ".check_names")]]
+SEXP check_names(List x, int column_count) {
+  CharacterVector column_names = NULL;
+
+  for (int i = 0; i < x.size(); i++) {
+    List row = x[i];
+
+    if (row.size() != column_count) {
+      stop("Item " + toString(i) + ", " +
+           "expected: " + toString(column_count) + " columns, " +
+           "received: " + toString(row.size()));
+    }
+
+    if (!Rf_isNull(row.names())) {
```

Comment:
I don't believe this works because `row.names()` returns a `NamesProxy` object. Also see http://stackoverflow.com/questions/24744802/rcpp-how-to-check-if-any-attribute-is-null